config/src/main/java/tech/pegasys/pantheon/config/CliqueConfigOptions.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.config;
```

Comment:
That was my initial inclination but two things persuaded me otherwise:

 1. Knowledge of the specific consensus algorithms that are available and how they fit into the config file is already outside of the specific consensus modules (see `PantheonController.fromConfig`).
 2. If we try to model the config as separate domains with each consensus implementation having it's config objects in separate modules, there's no way to expose those separate config objects from the top level `GenesisConfigOptions` without also exposing the inner workings of the config which is what we were trying to encapsulate in the first place.

Reply:
Those don't quite persuade me, but not going to block the improvement. Can revisit it later if the config starts getting in the way of the architecture requirements.